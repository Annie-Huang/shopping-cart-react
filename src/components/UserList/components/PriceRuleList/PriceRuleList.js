import React from 'react';
import PropTypes from 'prop-types';

export const PriceRuleList = ({rules, products}) => {

    const setPricingRulesForDisplay = (rules, products) => {
        let pricingRules = [];
        if (rules) {
            rules.forEach(rule => {
                let newPrice;
                const productname = products.find((product) => product.id === rule.productId).name;
                if (rule.ruleName === 'buyXQtyForYQtyPrice') {
                    const xQty = rule.attributes.find((attr) => attr.name === 'xQty').value;
                    const yQty = rule.attributes.find((attr) => attr.name === 'yQty').value;
                    pricingRules.push('Gets a **' + xQty + ' for ' + yQty + ' deal on ' + productname + 's**');
                } else if (rule.ruleName === 'newUnitPrice') {
                    newPrice = rule.attributes.find((attr) => attr.name === 'newPrice').value;
                    pricingRules.push('Gets a discount on **' + productname + 's where the price drops to $' + newPrice + ' per ad**');
                } else if (rule.ruleName === 'newUnitPriceWithMinQty') {
                    newPrice = rule.attributes.find((attr) => attr.name === 'newPrice').value;
                    const minQty = rule.attributes.find((attr) => attr.name === 'minQty').value;
                    pricingRules.push('Gets a discount on **' + productname + 's when ' + minQty +
                        ' or more** are purchased. The price drops to **$' + newPrice + ' per ad**');
                }
            });
        }
        return pricingRules;
    };

    const pricingRules = setPricingRulesForDisplay(rules, products).map((rule, index) =>
        <div className='col-md-12' key={index}>{rule}</div>
    );

    return (
        <div className='row'>
            {pricingRules}
        </div>
    );
};

PriceRuleList.propTypes = {
    rules: PropTypes.array.isRequired,
    products: PropTypes.array.isRequired,
};

export default PriceRuleList;
