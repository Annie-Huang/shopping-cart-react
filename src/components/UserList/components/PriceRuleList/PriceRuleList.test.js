import {shallow} from "enzyme/build";
import {PriceRuleList} from "./PriceRuleList";
import React from "react";


describe('Spinner', () => {
    it('should render Spinner component when loading', () => {
        const fordUser =  require('../../../../resources/fixtures/user-ford.json');
        const products =  require('../../../../resources/fixtures/products.json');
        const props = {
            rules: fordUser.pricingRules,
            products: products
        };
        expect(shallow(<PriceRuleList {...props} />)).toMatchSnapshot();
    });

});
