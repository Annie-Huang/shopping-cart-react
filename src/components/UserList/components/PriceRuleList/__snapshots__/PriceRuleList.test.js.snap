// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render Spinner component when loading 1`] = `
<div
  className="row"
>
  <div
    className="col-md-12"
    key="0"
  >
    Gets a **5 for 4 deal on Classic Ads**
  </div>
  <div
    className="col-md-12"
    key="1"
  >
    Gets a discount on **Standout Ads where the price drops to $309.99 per ad**
  </div>
  <div
    className="col-md-12"
    key="2"
  >
    Gets a discount on **Premium Ads when 3 or more** are purchased. The price drops to **$389.99 per ad**
  </div>
</div>
`;
